pkg/agent/entrypoint.go

```diff
@@ -25,17 +25,24 @@ type MainRunner struct {
 }
 
 func (r MainRunner) Run(ctx context.Context) error {
-	vmEvents := make(chan vmEvent)
-
 	buildInfo := util.GetBuildInfo()
 	klog.Infof("buildInfo.GitInfo:   %s", buildInfo.GitInfo)
 	klog.Infof("buildInfo.GoVersion: %s", buildInfo.GoVersion)
 
+	vmEventQueue := pubsub.NewUnlimitedQueue[vmEvent]()
+	defer vmEventQueue.Close()
```

Comment:
the queue doesn't have any background threads: closing the queue just means that it'll error if you try and add new things to the queue. 

Add will only ever error in two cases: 
- the queue is closed (this is the condition where a channel would panic)
- you are using a limited queue and it's at capacity. (retryable. I also made some interfaces to queues that let you iterate over results/have "force" or blocking ops that may make more sense in other cases.)

(both of these conditions can be checked with `errors.Is` and are probably not worth logging)

Reply:
Going to keep this, so that if we still have lots of things being added to the queue after shutdown, we have logs that that's what's going on